examples/custom_table.rs

```diff
@@ -1,19 +1,16 @@
 use std::convert::TryFrom;
 
 use multihash::derive::Multihash;
-use multihash::typenum::{U20, U25, U64};
 use multihash::{
-    Digest, Error, Hasher, MultihashDigest, MultihashGeneric, Sha2Digest, Sha2_256, Size,
-    StatefulHasher,
+    Digest, Error, Hasher, MultihashDigest, MultihashGeneric, Sha2Digest, Sha2_256, StatefulHasher,
 };
 
 // You can implement a custom hasher. This is a SHA2 256-bit hasher that returns a hash that is
 // truncated to 160 bits.
 #[derive(Default, Debug)]
 pub struct Sha2_256Truncated20(Sha2_256);
-impl StatefulHasher for Sha2_256Truncated20 {
-    type Size = U20;
-    type Digest = Sha2Digest<Self::Size>;
+impl StatefulHasher<20> for Sha2_256Truncated20 {
```

Comment:
this should still be an associated constant

Reply:
what's `Self::SIZE`?

Reply:
`Self::SIZE` replaced `size()` since while it could be a `const fn` it made more sense (in my mind) to make it a public associated constant. 

It would make sense to move back to `const fn size()` in the case of having `Size` be an associated constant without being repetitive.

Reply:
It seems using an assosicated const as a paramater is still part of the incomplete `const_generics` feature in nightly, though with enough feature flags we can make it look something like this 

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=30eed71199c4aecc0af8a0a893685084

Reply:
Not being able to have it as associate constant in current stable Rust sucks :-/

Reply:
are we in a hurry to merge this? it's a usability regression, and comparing it to the benefits, maybe we should hold off until more of const-generics is stabilized.

Reply:
I'm not in a hurry. Though I fear that it will take a long time until it's usable on stable.

@mriise What do you think about perhaps fixing everything up a single commit (once we are happy with the state of it) and we just keep it open? Then people who want const generic support can easily cherry-pick it themselves and it should also be easier to rebase it if we make other changes (e.g. fixing the blake3 hashing). This way all your work is not lost, still usable and hopefully find it's way in some day.

Reply:
Although it is disappointing, I would have to agree. I'll do the final bits of cleanup and request another review.